src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Pagination.scss';
+import Button from '../Button';
+import PaginationButton from './PaginationButton';
+
+const { block, elem } = bem({
+    name: 'Pagination',
+    classnames: styles,
+    propsToMods: ['align']
+});
+
+const Pagination = props => {
+    const {
+        align,
+        currentPage,
+        maxPageButtons,
+        totalPages,
+        prevLabel,
+        nextLabel,
+        lastLabel,
+        firstLabel,
+        onClick,
+        ...rest
+    } = props;
+
+    const defineRange = () => {
+        const showBefore = Math.floor((maxPageButtons - 1) / 2);
+        const start = Math.max(
+            2,
+            Math.min(currentPage - showBefore, totalPages - maxPageButtons + 1)
+        );
+        const end = Math.min(totalPages, start + maxPageButtons - 2);
+        const range = new Array(end - start + 1).fill().map((_, i) => start + i);
+
+        return range;
+    };
+
+    const isPrevDisabled = currentPage === 1;
+    const isNextDisabled = currentPage === totalPages;
+    const range = defineRange();
+
+    const handleClick = e => {
+        const {
+            dataset: { page = null }
+        } = e.target;
+        const pageNum = +page;
+
+        if (!onClick || !pageNum || pageNum === currentPage) {
+            return false;
+        }
+
+        return onClick(e, pageNum);
+    };
+
+    return (
+        <nav {...rest} {...block(props)} aria-label="pagination">
+            {!!firstLabel && (
+                <Button
+                    {...elem('button', props)}
+                    onClick={handleClick}
+                    disabled={isPrevDisabled}
+                    context="link"
+                    data-page={1}
+                    aria-disabled={isPrevDisabled}
+                >
+                    {`\u00ab ${firstLabel}`}
+                </Button>
+            )}
+            {!!prevLabel && (
+                <Button
+                    {...elem('button', props)}
+                    onClick={handleClick}
+                    disabled={isPrevDisabled}
+                    context="link"
+                    data-page={currentPage - 1}
+                    aria-disabled={isPrevDisabled}
+                >
+                    {`\u2039 ${prevLabel}`}
+                </Button>
+            )}
+            <PaginationButton
+                onClick={handleClick}
+                isActive={currentPage === 1}
+                data-page={1}
+                key="page_1"
+            >
+                {1}
+            </PaginationButton>
+            {range[0] > 2 && <div {...elem('gap', props)}>&hellip;</div>}
```

Comment:
I'm missing the `&hellip;` between `current ... next> last>>`

Reply:
Because it's not there in the design.

Reply:
> When some pages are not listed, show a 'gap' (horizontal ellipsis, see design attached)

It doesn't specify which unlisted pages are being referred to, and there is no design attached.

I read this as including _'pages not listed between current and last'_.

Reply:
The designs are both on conceptboard as well as in the story.

What I mean with 'some pages are not listed' are those in between the first page and the current. As you can see in the design there is no gap beyond that, nor is the last page always shown.